pennylane/tape/reversible.py

```diff
@@ -189,10 +189,10 @@ def reversible_diff(self, idx, params, **options):
 
         if op.name == "Rot":
             decomp = op.decomposition()
-            generator, multiplier = qml.utils.get_generator(decomp[p_idx])
+            multiplier, generator = qml.generator(decomp[p_idx])
             between_ops = decomp[p_idx + 1 :] + between_ops

```

Comment:
This is something we discussed a lot internally, and went back and forth a _lot_ (CCing in @dwierichs for interest).

At one stage in the process, we considered having
```pycon
>>> qml.Rot(0.1, 0.2, 0.3, wires=0).generator()
[<Hamiltonian>, <Hamiltonian>, <Hamiltonian>]
```
e.g., one 'generator' per parameter.

But we decided not to in the end, iirc, as this felt like abuse of nomenclature, since the concept of an operation generator is not well defined for multiple parameters

Reply:
I agree. For `Rot` there is a clear case to be made that the three generators could be used. However, `Rot` really just is an abbreviation for the single rotations and a decomposition into those (in order to unable generator-unlocked features) does not change much.
Adding lists of generators might tempt users to add generators to multi-parameter gates that are not a simple concatenation of single-parameter gates.